api/getAllIds.tsx

```diff
@@ -0,0 +1,28 @@
+import {useEffect, useState} from "react";
+import MuseumsResponse from "../model/MuseumsResponse";
+
+export default function getAllIds(search: string): Action<Array<number>> {
+
+    const [inProgress, setInProgress] = useState(true);
+    const [museumResponse, setMuseumResponse] = useState<MuseumsResponse>();
+
+    function action() {
+        const baseLink = 'https://collectionapi.metmuseum.org/public/collection/v1/objects'
+        const searchLink = `https://collectionapi.metmuseum.org/public/collection/v1/search?q=${search.trim()}`
+
+        setInProgress(true);
+        fetch(search.trim().length == 0 ? baseLink : searchLink)
+            .then((res) => res.json())
+            .then((resp) => setMuseumResponse(resp))
+            .catch(console.log)
+            .finally(() => setInProgress(false))
+    }
+
+    useEffect(action, []);
```

Comment:
For the cleanest solution we should wrap the `action` function in the [`useCallback` hook](https://react.dev/reference/react/useCallback). Also, I would rename the function to make clear what it does. In this case the code could look moreless like this:

```suggestion
    const fetchMuseumObjects = useCallback(() => {
        const baseLink = 'https://collectionapi.metmuseum.org/public/collection/v1/objects'
        const searchLink = `https://collectionapi.metmuseum.org/public/collection/v1/search?q=${search.trim()}`

        setInProgress(true);
        fetch(search.trim().length == 0 ? baseLink : searchLink)
            .then((res) => res.json())
            .then((resp) => setMuseumResponse(resp))
            .catch(console.log)
            .finally(() => setInProgress(false))
    }, [setMuseumResponse, setInProgress])

    useEffect(fetchMuseumObjects, [fetchMuseumObjects]);
```